src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -406,13 +436,13 @@ private bool ShortNameArgumentMatch(string arg, string shortName)
             return arg.Equals(shortName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        private void ImportType(Type type, ArrayList mappings, ArrayList importedTypes, bool verbose, XmlReflectionImporter importer, bool parsableerrors)
+        private void ImportType(Type type, string defaultNamespace, ArrayList mappings, ArrayList importedTypes, bool verbose, XmlReflectionImporter importer, bool parsableerrors)
         {
             XmlTypeMapping xmlTypeMapping = null;
-            var localImporter = new XmlReflectionImporter();
+            var localImporter = new XmlReflectionImporter(defaultNamespace);
             try
             {
-                xmlTypeMapping = localImporter.ImportTypeMapping(type);
+                xmlTypeMapping = localImporter.ImportTypeMapping(type, defaultNamespace);
```

Comment:
Isn't this redundant, given that localImporter was constructed with a default namespace?

Reply:
If you look at XmlSerializer.xs (which is used by the framework when generating the assembly on runtime), you will observe that the default namespace is set both in the XmlReflectionImporter constructor and in the call to ImportTypeMapping.
Due to the way XmlReflectionImporter is designed, providing the 'defaultNamespace' argument to ImportTypeMapping is not redundant (omitting the value caused some of my tests to fail).